src/nhs_grid.jl

```diff
@@ -36,10 +36,20 @@ since not sorting makes our implementation a lot faster (although less paralleli
                             [`PeriodicBox`](@ref).
 - `cell_list`:              The cell list that maps a cell index to a list of points inside
                             the cell. By default, a [`DictionaryCellList`](@ref) is used.
-- `threaded_update = true`: Can be used to deactivate thread parallelization in the
-                            neighborhood search update. This can be one of the largest
-                            sources of variations between simulations with different
-                            thread numbers due to neighbor ordering changes.
+- `update_strategy = nothing`: Strategy to parallelize `update!`. Available options are:
+    - `nothing`:        Automatically choose the best available option.
+    - `:parallel`:      Fully parallel update by using atomic operations to avoid race
+                        conditions when adding points into the same cell. This is not
+                        available for all cell list implementations, but is the default when
+                        available.
+    - `:semi-parallel`: Loop over all cells in parallel to mark cells with points that now
+                        belong to a different cell. Then, move points of affected cells
+                        serially. This is available for all cell list implementations and is
+                        the default when `:parallel` is not available.
+    - `:serial`:        Deactivate parallelization in the neighborhood search update.
+                        Parallel update can be one of the largest sources of variations
+                        between simulations with different thread numbers due to neighbor
+                        ordering changes.
```

Comment:
```suggestion
    - `:serial`:        Deactivate parallelization in the neighborhood search update.
                        Parallel update can be one of the largest sources of runtime variations
                        between simulations with different thread numbers due to neighbor
                        ordering changes.
```

Reply:
"one of the largest sources of *error variations* between simulations with different thread numbers"?

Reply:
Ah, then I misread it. Either way, it was not clear before what kind of variations we're talking about here, thus it would be great if you can clarify it